### PR TITLE
Feat/equity

### PR DESCRIPTION
**Pull request by IanAtCredo**
## Change description

**Equity Module**
It works like so:
* Calculate descriptive statistics (average value per group)
* Calculate top-level statistic to evaluate whether groups differ
      * If groups differ, calculates posthoc tests to tell which groups differ.
* Statistics (top-level and posthoc) differ depending on the outcome type:
     * binary: chisq
     * continuous: anova, followed by tukey hsd
     * bounded [0-1] continuous (like proportion). Transform to logit than treated as continuous above

**Issues this module brought up**
* I didn't build this module to run over multiple sensitive features within the module. Instead, I am refactoring everything so that modules only work on one sensitive feature. I found having to loop over sensitive features in every function very cumbersome. PR following!
* There are a number of things we'd like to report that aren't metric. We will move to reporting "tables" to accommodate this different kind of data
* Statistics aren't really metrics, and should have their own structure as well

### CODE REVIEW

**Review by amrasekh — APPROVED** [review]

File: credoai/modules/general_modules/equity.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+import numpy as np
+import pandas as pd
+from credoai.modules.credo_module import CredoModule
+from credoai.utils import NotRunError, ValidationError
+from scipy.stats import chi2_contingency, chisquare, f_oneway, tukey_hsd
+from sklearn.utils.multiclass import type_of_target
+
+
+class EquityModule(CredoModule):
```
[the remark this answers is not in the document]
Comment (IanAtCredo) [reply credoai/modules/general_modules/equity.py:11]: This is really aimed at outcomes. It's somewhat overlapping with the fairnessmodule, but different enough that it seemed worth making a different module. I think modules should be as "single purpose" as possible, with assessments potentially becoming more multipurpose.
